File: app/src/main/java/com/sjl/bookmark/ui/contract/CheckLockContract.java
package com.sjl.bookmark.ui.contract;

import com.sjl.bookmark.widget.LockPatternView;
import com.sjl.core.mvp.BaseContract;
import com.sjl.core.mvp.BasePresenter;

import java.util.List;

/**
 * TODO
 *
 * @author Kelly
 * @version 1.0.0
 * @filename CheckLockContract.java
 * @time 2018/11/26 11:33
 * @copyright(C) 2018 song
 */
public interface CheckLockContract {
    /**
     * 图案锁验证view
     */
    interface View extends BaseContract.IBaseView {

        void lockDisplayError();

        void kill();
    }

    abstract class Presenter extends BasePresenter<View> {
        public abstract void check(List<LockPatternView.Cell> pattern);
    }
}

File: app/src/main/java/com/sjl/bookmark/ui/contract/CheckLockContract.kt
package com.sjl.bookmark.ui.contract

import com.sjl.bookmark.widget.LockPatternView
import com.sjl.core.mvp.BaseContract.IBaseView
import com.sjl.core.mvp.BasePresenter

/**
 * TODO
 *
 * @author Kelly
 * @version 1.0.0
 * @filename CheckLockContract.java
 * @time 2018/11/26 11:33
 * @copyright(C) 2018 song
 */
interface CheckLockContract {
    /**
     * 图案锁验证view
     */
    interface View : IBaseView {
        fun lockDisplayError()
        fun kill()
    }

    abstract class Presenter : BasePresenter<View>() {
        abstract fun check(pattern: List<LockPatternView.Cell>)
    }
}